src/java/org/apache/cassandra/tcm/RemoteProcessor.java

```diff
@@ -152,34 +150,37 @@ public ClusterMetadata fetchLogAndWait(Epoch waitFor, Retry.Deadline retryPolicy
 
     public static ClusterMetadata fetchLogAndWait(CandidateIterator candidateIterator, LocalLog log)
     {
-        Promise<LogState> remoteRequest = new AsyncPromise<>();
-        return fetchLogAndWaitInternal(remoteRequest, candidateIterator, log);
+        try
+        {
+            return fetchLogAndWaitInternal(candidateIterator, log).awaitUninterruptibly().get();
```

Comment:
Should we `awaitThrowUncheckedOnInterrupt`, or even just `await` here? I'm thinking of tests really where the origin implementation (before CASSANDRA-19514) could cause occasional timeouts.

Reply:
Do you mean `await` with a timeout, or without? My logic for doing it this way was that we will rely on messaging service timeouts now to time this one out, so it should never be indefinite.

Reply:
fair enough, I was concerned about in-flight requests preventing a timely shutdown, but this will now be bounded by `new Retry.Backoff(TCMMetrics.instance.fetchLogRetries));`, right?